static/local.js

```diff
@@ -133,4 +135,311 @@ function init() {
+  // 3. Converted result pushed to index of points depending on color (0: red, 1: green, 2: blue, 3: yellow) and so on. [x]
+  // 4. If we stop on colored stuff, We don't push info and let player re-connect by re-sticking lines. [x]
+  // 5. Else, Push info and draw line with rounded corner from first pos to second pos and so on. [x]
+  // 6. Check for connecting point with point at end. [x]
+  
+  // TODO:
+  // 1. Disallow diagonal movement (Without corrupt connecting).
+  // 2. Reset line if collision with other lines of other colors or with same line itself.
+  // 3. Better way to detect victory!
+  var lines = [ [], [], [], [] ];
+  
+  var colors_points = [
+    { start: { i: 0, j: 0 }, end: { i: 4, j: 0 } }, // red
+    { start: { i: 0, j: 2 }, end: { i: 4, j: 1 } }, // green
+    { start: { i: 0, j: 3 }, end: { i: 4, j: 2 } }, // blue
+    { start: { i: 0, j: 4 }, end: { i: 4, j: 4 } }, // yellow
+  ];
+
+  // 0: none, 1: red, 2: green, 3: blue, 4: yellow
+  // 5x5 rendered
+  var grid = [
+    [ 1, 0, 2, 3, 4 ],
+    [ 0, 0, 0, 0, 0 ],
+    [ 0, 0, 0, 0, 0 ],
+    [ 0, 0, 0, 0, 0 ],
+    [ 1, 2, 3, 0, 4 ],
+  ];
+  
+  function draw_grid(i, j) {
+    power_game_context.strokeStyle = "gray";
+    power_game_context.fillStyle = "black";
+    power_game_context.fillRect(j * 50, i * 50, 50, 50);
+    power_game_context.strokeRect(j * 50, i * 50, 50, 50);
+  }
+  
+  function draw_connection_points(i, j) {
+    if (is_connection_point(i, j)) {
+      if (grid[i][j] != 0) {
+        power_game_context.beginPath();
+        power_game_context.arc((j * 50) + 25, (i * 50) + 25, 25, 90, 180 * Math.PI); // Better?
+        power_game_context.closePath();
+        power_game_context.fill();
+      }
+    }
+  }
+  
+  function draw_connection_lines(lines) {
+    for (var i = 0; i < lines.length; i++) {
+      if (i == 0) {
+        power_game_context.strokeStyle = "red";
+      } else if (i == 1) {
+        power_game_context.strokeStyle = "green";
+      } else if (i == 2) {
+        power_game_context.strokeStyle = "blue";
+      } else if (i == 3) {
+        power_game_context.strokeStyle = "yellow";
+      } else {
+        power_game_context.strokeStyle = "blank";  // Don't draw anything else!
+      }
+      
+      for (var j = 1; j < lines[i].length; j++) {
+        var p1 = point_pos(lines[i][j - 1].i, lines[i][j - 1].j);
+        var p2 = point_pos(lines[i][j].i, lines[i][j].j);
+        
+        // Rounded connection lines
+        power_game_context.lineWidth = 15;
+        power_game_context.lineJoin = "round";
+        power_game_context.beginPath();
+        power_game_context.moveTo(p1.x + 25, p1.y + 25);
+        power_game_context.lineTo(p2.x + 25, p2.y + 25);
+        power_game_context.closePath();
+        power_game_context.stroke();
+        power_game_context.lineWidth = 1;
+      }
+    }
+  }
+
+  function render() {
```

Comment:
This works well, but doesn't seem to fill the entire canvas.
Instead of doing * 50 do `canvas.width/5` that will make sure it still works if we resize the canvas

![image](https://user-images.githubusercontent.com/7799288/116594530-2a892c80-a8d7-11eb-9e28-ae5836d77ff2.png)